sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3290,10 +3368,17 @@ pi_result piMemRelease(pi_mem Mem) {
     } else {
       auto Buf = static_cast<_pi_buffer *>(Mem);
       if (!Buf->isSubBuffer()) {
-        if (enableBufferPooling()) {
-          PI_CALL(piextUSMFree(Mem->Context, Mem->getZeHandle()));
+        if (Mem->getHostPtrImported()) {
+          ze_driver_handle_t driverHandle =
+              Mem->Context->Devices[0]->Platform->ZeDriver;
```

Comment:
why do you query "0" device?

Reply:
maybe we should remember the device handle when importing?

Reply:
The import is not really device-specific. Host memory is imported into host USM. This is just a way to get the L0 driver handle, which the API seems to need.